diff --git a/dist/index.d.mts b/dist/index.d.mts
index 8b23996dcce6c1ad5b17ef59f92196fb97312d79..80be2e52a347042b89da8e502834afd92120877a 100644
--- a/dist/index.d.mts
+++ b/dist/index.d.mts
@@ -10,13 +10,7 @@ declare const groqLanguageModelOptions: z.ZodObject<{
         raw: "raw";
         hidden: "hidden";
     }>>;
-    reasoningEffort: z.ZodOptional<z.ZodEnum<{
-        none: "none";
-        default: "default";
-        low: "low";
-        medium: "medium";
-        high: "high";
-    }>>;
+    reasoningEffort: z.ZodOptional<z.ZodString>;
     parallelToolCalls: z.ZodOptional<z.ZodBoolean>;
     user: z.ZodOptional<z.ZodString>;
     structuredOutputs: z.ZodOptional<z.ZodBoolean>;
diff --git a/dist/index.d.ts b/dist/index.d.ts
index 8b23996dcce6c1ad5b17ef59f92196fb97312d79..80be2e52a347042b89da8e502834afd92120877a 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -10,13 +10,7 @@ declare const groqLanguageModelOptions: z.ZodObject<{
         raw: "raw";
         hidden: "hidden";
     }>>;
-    reasoningEffort: z.ZodOptional<z.ZodEnum<{
-        none: "none";
-        default: "default";
-        low: "low";
-        medium: "medium";
-        high: "high";
-    }>>;
+    reasoningEffort: z.ZodOptional<z.ZodString>;
     parallelToolCalls: z.ZodOptional<z.ZodBoolean>;
     user: z.ZodOptional<z.ZodString>;
     structuredOutputs: z.ZodOptional<z.ZodBoolean>;
diff --git a/dist/index.js b/dist/index.js
index 45a104f2e0775761858eac2a82ced64bceba1f5e..f60ac36f4a064d527e8f8881b1d6c58ff69286a3 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -214,7 +214,7 @@ var groqLanguageModelOptions = import_v4.z.object({
    * Specifies the reasoning effort level for model inference.
    * @see https://console.groq.com/docs/reasoning#reasoning-effort
    */
-  reasoningEffort: import_v4.z.enum(["none", "default", "low", "medium", "high"]).optional(),
+  reasoningEffort: import_v4.z.string().optional(),
   /**
    * Whether to enable parallel function calling during tool use. Default to true.
    */
diff --git a/dist/index.mjs b/dist/index.mjs
index c644c32235d8fa88c51c0fc6958feb1da4877c96..2c2f81869673eb4633e843d93ff5376abf1e67d0 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -203,7 +203,7 @@ var groqLanguageModelOptions = z.object({
    * Specifies the reasoning effort level for model inference.
    * @see https://console.groq.com/docs/reasoning#reasoning-effort
    */
-  reasoningEffort: z.enum(["none", "default", "low", "medium", "high"]).optional(),
+  reasoningEffort: z.string().optional(),
   /**
    * Whether to enable parallel function calling during tool use. Default to true.
    */
diff --git a/src/groq-chat-options.ts b/src/groq-chat-options.ts
index 3812cdf53308709f166f05c58c5d46a5d8189c8b..af520c5459bd752b3cce03c3b4afbeed31157d90 100644
--- a/src/groq-chat-options.ts
+++ b/src/groq-chat-options.ts
@@ -33,9 +33,7 @@ export const groqLanguageModelOptions = z.object({
    * Specifies the reasoning effort level for model inference.
    * @see https://console.groq.com/docs/reasoning#reasoning-effort
    */
-  reasoningEffort: z
-    .enum(['none', 'default', 'low', 'medium', 'high'])
-    .optional(),
+  reasoningEffort: z.string().optional(),
 
   /**
    * Whether to enable parallel function calling during tool use. Default to true.
